x-pack/plugins/endpoint/server/routes/endpoints.test.ts

```diff
@@ -0,0 +1,183 @@
+import * as data from '../test_data/all_endpoints_data.json';
+
+describe('test endpoint route', () => {
+  let routerMock: jest.Mocked<IRouter>;
+  let mockResponse: jest.Mocked<KibanaResponseFactory>;
+  let mockClusterClient: jest.Mocked<IClusterClient>;
+  let mockScopedClient: jest.Mocked<IScopedClusterClient>;
+  let routeHandler: RequestHandler<any, any, any>;
+  let routeConfig: RouteConfig<any, any, any, any>;
+
+  beforeEach(() => {
+    mockClusterClient = elasticsearchServiceMock.createClusterClient() as jest.Mocked<
+      IClusterClient
+    >;
+    mockScopedClient = elasticsearchServiceMock.createScopedClusterClient();
+    mockClusterClient.asScoped.mockReturnValue(mockScopedClient);
+    routerMock = httpServiceMock.createRouter();
+    mockResponse = httpServerMock.createResponseFactory();
+    registerEndpointRoutes(routerMock, {
+      logFactory: loggingServiceMock.create(),
+      config: () => Promise.resolve(EndpointConfigSchema.validate({})),
+    });
+  });
+
+  it('test find the latest of all endpoints', async () => {
+    const mockRequest = httpServerMock.createKibanaRequest({
+      body: {},
+      params: {},
+    });
+
+    const response: SearchResponse<EndpointData> = (data as unknown) as SearchResponse<
+      EndpointData
+    >;
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() => Promise.resolve(response));
+    [routeConfig, routeHandler] = routerMock.get.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/endpoints')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          elasticsearch: {
+            dataClient: mockScopedClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+
+    expect(mockScopedClient.callAsCurrentUser).toBeCalledWith('search', {
+      from: 0,
+      size: 10,
+      body: {
+        query: {
+          match_all: {},
+        },
+        collapse: {
+          field: 'machine_id',
+          inner_hits: {
+            name: 'most_recent',
+            size: 1,
+            sort: [{ created_at: 'desc' }],
+          },
+        },
+        aggs: {
+          total: {
+            cardinality: {
+              field: 'machine_id',
+            },
+          },
+        },
+        sort: [
+          {
+            created_at: {
+              order: 'desc',
+            },
+          },
+        ],
+      },
+      index: 'endpoint-agent*',
+    });
+    expect(routeConfig.options).toEqual({ authRequired: true });
+    expect(mockResponse.ok).toBeCalled();
+    const endpointResultList = mockResponse.ok.mock.calls[0][0]?.body as EndpointResultList;
+    expect(endpointResultList.endpoints.length).toEqual(3);
+    expect(endpointResultList.total).toEqual(3);
+    expect(endpointResultList.requestIndex).toEqual(0);
+    expect(endpointResultList.requestPageSize).toEqual(10);
+  });
+
+  it('test find the latest of all endpoints with params', async () => {
```

Comment:
Should we add a test with negative numbers/floats? Or does the schema enforce this?

Reply:
> Should we add a test with negative numbers/floats? Or does the schema enforce this?

We can enforce it in the schema.

Reply:
@nnamdifrankie are we enforcing it or are you saying we could?

Reply:
It is enforced by the schema  https://github.com/elastic/kibana/pull/53861/files#diff-fb44c643877621f0f4e2e866c3099c52R49